### PR TITLE
[12.0][MIG] migration module privacy

### PR DESCRIPTION
**Pull request by njeudy**
- Juste change version and test installation

WIP: 

- Clean form view

**Before:**
capture d ecran 2019-01-26 10 39 36
**After:**
image

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +25,7 @@ addons:
 
 env:
   global:
-  - VERSION="10.0" TESTS="0" LINT_CHECK="0" TRANSIFEX="0"
+  - VERSION="12.0" TESTS="0" LINT_CHECK="0" TRANSIFEX="0"
```
Comment (pedrobaeza) [inline .travis.yml]: Please use latest template with `MAKEPOT` and one less build.

File: privacy/__manifest__.py [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
 {
     'name': 'Data Privacy and Protection',
-    'version': '11.0.1.0.0',
+    'version': '12.0.1.0.0',
     'category': 'Data Protection',
     'summary': 'Provides data privacy and protection features '
                'to comply to regulations, such as GDPR.',
     'author': "Eficent, "
               "Tecnativa, "
+              "Nicolas JEUDY, "
```
Comment (pedrobaeza) [inline privacy/__manifest__.py]: These little changes don't grant you as co-author. Please only add you on contributors.

File: privacy/views/privacy_activity_view.xml [hunk 3]
```diff
@@ -33,28 +33,34 @@
...
+                              <field
+                                  name="subject_domain"
+                                  widget="domain"
+                                  options='{"model": "res.partner"}'
+                                  attrs='{"required": [("subject_find", "=", True)],
+                                          "invisible": [("subject_find", "=", False)]}'
+                              />
+                            </group>
+                        </page>
+                        <page string="Processors" name="processors">
```
Comment (pedrobaeza) [inline privacy/views/privacy_activity_view.xml]: Doing a page only for this field is too much IMO.

File: privacy/views/privacy_activity_view.xml [hunk 4]
```diff
@@ -33,28 +33,34 @@
...
                         </group>
                     </group>
                     <notebook string="Details" name="advanced">
-                        <page string="Description">
+                        <page string="Description" name="description">
                             <group>
                                 <field name="description" nolabel="1"/>
                             </group>
                         </page>
+                        <page string="Subjects" name="subjects">
```
Comment (pedrobaeza) [inline privacy/views/privacy_activity_view.xml]: They are only 2 fields. No need of extra page for that.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: privacy/__manifest__.py (hunk 2 above)
Comment (njeudy) [reply privacy/__manifest__.py]: ok, is there a place to now when you become author ?

File: privacy/views/privacy_activity_view.xml (hunk 3 above)
Comment (njeudy) [reply privacy/views/privacy_activity_view.xml]: This is the base module but some other will come and will have more option for processors, like tracking if they are GPDR compliant, extra EU, etc ..
That's why I proposed this.

File: privacy/views/privacy_activity_view.xml (hunk 4 above)
Comment (njeudy) [reply privacy/views/privacy_activity_view.xml]: @pedrobaeza for now yes but will extend with some modules. For me privacy is a base module, but will add many dependent module to have a complete activity description and management.

File: privacy/__manifest__.py (hunk 2 above)
Comment (pedrobaeza) [reply privacy/__manifest__.py]: There's no exact metric, and it's valued by reviewers on each specific case. A migration hardly will be a co-author task, as most of the work is to just test things. If the migration requires to change a significant part of the module, then can be qualified for it. Also introducing a new big feature, doing a big refactoring, or adding a big bunch of tests are other promoting tasks.

File: privacy/views/privacy_activity_view.xml (hunk 3 above)
Comment (pedrobaeza) [reply privacy/views/privacy_activity_view.xml]: I don't like too much any way. What original promoters of this module - @Yajo @mreficent - think?

File: privacy/views/privacy_activity_view.xml (hunk 4 above)
Comment (pedrobaeza) [reply privacy/views/privacy_activity_view.xml]: The same open question to the rest of the reviewers

Comment (njeudy) [reply privacy/views/privacy_activity_view.xml]: I can make without page, and use separators, but think we have to organise data by type (description, subjects, processors) and have a clean small form on first with a resume of the activity.

File: privacy/views/privacy_activity_view.xml (hunk 3 above)
Comment (njeudy) [reply privacy/views/privacy_activity_view.xml]: For exemple I need to add purpose, storage medium used, if processors are Extra EU, need to track legal documents etc ..

I understand your comment, think we can have a good solution.

File: privacy/views/privacy_activity_view.xml (hunk 4 above)
Comment (ghost) [reply privacy/views/privacy_activity_view.xml]: i would prefer a extra page

**Comment by yajo** [comment]
What if we open a "Detials" page, where "Subjects" is a subsection, and "Processors" is another one, and more sections can be added or removed?

File: privacy/views/privacy_activity_view.xml (hunk 4 above)
Comment (njeudy) [reply privacy/views/privacy_activity_view.xml]: @Axel-G what do you mean ? can you explain ?

**Comment by njeudy** [comment]
> What if we open a "Detials" page, where "Subjects" is a subsection, and "Processors" is another one, and more sections can be added or removed?

Hum, think we can do this but details will be a big page because I need to add some details:

- Subjects: add res.partner list if in odoo, add subject consent link (action button I think) to all consents from partners, 
- Processors: add processors list (action button ?) and manage data and documents associated with.

"Where processing is to be carried out on behalf of a controller, the controller shall use only processors providing sufficient guarantees to implement appropriate technical and organisational measures in such a manner that processing will meet the requirements of this Regulation and ensure the protection of the rights of the data subject"

I wish to manage this on each activity. So activities form can be complex.

Here is a type of document I wish to achieve for each activities (in french sorry):

[registre_rgpd_basique.pdf](https://github.com/OCA/data-protection/files/2804804/registre_rgpd_basique.pdf)

**Comment by yajo** [comment]
Thanks for the doc.

The kind of data I think can be organizated as follows:

1. Basic activity information:
   - Activity name
   - Subjects
   - Controller
   - Processors
   - Description
1. Info about contents of the processed data:
   - Objectives
   - Subject Categories
   - Data categories
   - Transfer out of UE
1. Data security:
   - Measures
   - Breaches
   - Audits
   - Etc.
1. Consent (this one comes from [v10's `privacy_consent`](https://github.com/OCA/data-protection/tree/10.0/privacy_consent), which I encourage you to take a look into if you didn't yet :wink:, and it's what concerns me most personally).
   - Wether to ask for it or not.
   - Default acceptance status
   - Email template
   - Consents sent/received
   - Other configs

Also, besides our comfortability on putting fields somewhere, I want to think on user's comfortability, which means having most important info available at first sight, and a form structure that avoids repeating information.

As a good base addon, this one only provides info contained in point 1. IMHO all that view should be arranged as it was previously.

I.e., for a user, not seeing the subjects or processors until hitting pages 2 and 3 seems like that info is not so important, but it is actually very basic.

You would probably need to add a new tab for point 2 and another for point 3. I also think that point 2 should appear before the description tab, just to make sure the user doesn't fill the description with things that he will have to write again in that other tab.

Point 4 already creates its own page, and it's OK as it is.

Well, this is my POV, and I just ask you to consider it, but I'm not gonna block the PR just because of this! :blush:

**Review by yajo — APPROVED** [review]

**Comment by njeudy** [comment]
ok thanks @Yajo , will keep it on first sight, but add some separators to have a clean form :)

**Comment by njeudy** [comment]
@pedrobaeza @Yajo I update screenshot for after ? remove notebook and add separator.

Is it ok for you ? Will push code if ok :)

**Comment by yajo** [comment]
Thanks! I like it more. However, I'd still put the description field into its own tab. Just to have a good place where to inject more tabs if needed.

For the base addon, the UX remains the same as there will be only 1 tab. For other addons, this lets them decide if their new fields should have a brand new tab or use one of the provided sections.

**Comment by njeudy** [comment]
@Yajo Done ! 
@pedrobaeza Any requests ?

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash your last 2 commits for merging this.

**Comment by njeudy** [comment]
@pedrobaeza done ! thanks for the review

**Pull request merged**